aria/parser/modeling/storage.py

```diff
@@ -0,0 +1,184 @@
+            s_node = create_node(s_service_instance, s_node_template, node)
+            model_storage.node.put(s_node)
+            create_interfaces(context, model_storage, node.interfaces, s_node, 'node', '_dry_node')
+
+    # Create relationships between nodes
+    for source_node in context.modeling.instance.nodes.itervalues():
+        for relationship in source_node.relationships:
+            s_source_node = model_storage.node.get_by_name(source_node.id)
+            s_target_node = model_storage.node.get_by_name(relationship.target_node_id)
+            s_relationship = create_relationship(s_source_node, s_target_node)
+            model_storage.relationship.put(s_relationship)
+            create_interfaces(context, model_storage, relationship.source_interfaces,
+                              s_relationship, 'relationship', '_dry_relationship')
+
+
+def create_service_template(context):
+    now = datetime.utcnow()
+    main_file_name = unicode(context.presentation.location)
+    try:
+        name = context.modeling.model.metadata.values.get('template_name')
+    except AttributeError:
+        name = None
+    return model.ServiceTemplate(
+        name=name or main_file_name,
+        description=context.modeling.model.description or '',
+        created_at=now,
+        updated_at=now,
+        main_file_name=main_file_name)
+
+
+def create_service_instance(context, service_template, service_instance_id):
+    now = datetime.utcnow()
+    return model.ServiceInstance(
+        name='%s_%s' % (service_template.name, service_instance_id),
+        service_template=service_template,
+        description=context.modeling.instance.description or '',
+        created_at=now,
+        updated_at=now)
+
+
+def create_node_template(service_template, node_template):
+    return model.NodeTemplate(
+        name=node_template.name,
+        type_name=node_template.type_name,
+        default_instances=node_template.default_instances,
+        min_instances=node_template.min_instances,
+        max_instances=node_template.max_instances or 100,
+        service_template=service_template)
+
+
+def create_node(service_instance, node_template, node):
+    return model.Node(
+        name=node.id,
+        state='',
+        node_template=node_template,
+        service_instance=service_instance)
+
+
+def create_relationship(source_node, target_node):
+    return model.Relationship(
+        source_node=source_node,
+        target_node=target_node)
+
+
+def create_interfaces(context, model_storage, interfaces, obj, attr_name, fn_name):
+    for interface_name, interface in interfaces.iteritems():
+        s_interface = model.Interface(name=interface_name,
+                                      type_name=interface.type_name)
+        setattr(s_interface, attr_name, obj)
+        try:
+            obj.source_interfaces.append(s_interface)
+        except AttributeError:
+            # is a node
```

Comment:
If it's a node it just doesn't have this extra constraint. Line #111 would be enough.